src/Adapters/Auth/AuthAdapter.js

```diff
@@ -1,20 +1,102 @@
 /*eslint no-unused-vars: "off"*/
+
+/**
+ * @interface ParseAuthResponse
+ * @property {Boolean} [doNotSave] If true, Parse Server will do not save provided authData.
+ * @property {Object} [response] If set, Parse Server will send the provided response to the client under authDataResponse
+ * @property {Object} [save] If set, Parse Server will save the object provided into this key, instead of client provided authData
+ */
+
+/**
+ * AuthPolicy
+ * default: can be combined with ONE additional auth provider if additional configured on user
+ * additional: could be only used with a default policy auth provider
+ * solo: Will ignore ALL additional providers if additional configured on user
+ * @typedef {"default" | "additional" | "solo"} AuthPolicy
```

Comment:
Declare this new type so it can be referenced properly with auto-completion:

```js

/**
 * The auth policy.
 * 
 * @typedef {'default' | 'additional' | 'solo'} AuthPolicy
 */
const AuthPolicy = Object.freeze({
  /** Auth provider can be combined with one additional auth provider, if there is an additional one configures for the user. */
  default: 'default',
  additional: 'additional',
  solo: 'solo',
});

class AuthAdapter {
  /**
   * @param {AuthPolicy} policy
   */
  constructor(policy = AuthPolicy.default) {
    if (![...Object.values(AuthPolicy)].includes(policy)) {
      // error
    }
  }
}
```